StripePaymentSheet/StripePaymentSheet/Source/PaymentSheet/ViewControllers/UpdatePaymentMethodViewController.swift

```diff
@@ -0,0 +1,256 @@
+//
+//  UpdatePaymentMethodViewController.swift
+//  StripePaymentSheet
+//
+//  Created by Nick Porter on 10/27/23.
+//
+import Foundation
+@_spi(STP) import StripeCore
+@_spi(STP) import StripePayments
+@_spi(STP) import StripePaymentsUI
+@_spi(STP) import StripeUICore
+import UIKit
+
+@MainActor
+protocol UpdatePaymentMethodViewControllerDelegate: AnyObject {
+    func didRemove(viewController: UpdatePaymentMethodViewController, paymentMethod: STPPaymentMethod)
+    func didUpdate(viewController: UpdatePaymentMethodViewController,
+                   paymentMethod: STPPaymentMethod,
+                   updateParams: STPPaymentMethodUpdateParams) async throws
+    func didDismiss(_: UpdatePaymentMethodViewController)
+}
+
+/// For internal SDK use only
+@objc(STP_Internal_UpdatePaymentMethodViewController)
+final class UpdatePaymentMethodViewController: UIViewController {
+    private let removeSavedPaymentMethodMessage: String?
+    private let isTestMode: Bool
+    private let viewModel: UpdatePaymentMethodViewModel
+
+    private var latestError: Error? {
+        didSet {
+            errorLabel.text = latestError?.localizedDescription
+            errorLabel.isHidden = latestError == nil
+        }
+    }
+
+    weak var delegate: UpdatePaymentMethodViewControllerDelegate?
+
+    // MARK: Navigation bar
+    internal lazy var navigationBar: SheetNavigationBar = {
+        let navBar = SheetNavigationBar(isTestMode: isTestMode,
+                                        appearance: viewModel.appearance)
+        navBar.delegate = self
+        navBar.setStyle(navigationBarStyle())
+        return navBar
+    }()
+
+    // MARK: Views
+    lazy var formStackView: UIStackView = {
+        let stackView = UIStackView(arrangedSubviews: [headerLabel, paymentMethodForm])
+        stackView.isLayoutMarginsRelativeArrangement = true
+        stackView.axis = .vertical
+        stackView.setCustomSpacing(16, after: headerLabel) // custom spacing from figma
+        if let footnoteLabel = footnoteLabel {
+            stackView.addArrangedSubview(footnoteLabel)
+            stackView.setCustomSpacing(8, after: paymentMethodForm) // custom spacing from figma
+            stackView.setCustomSpacing(32, after: footnoteLabel) // custom spacing from figma
+        }
+        else {
+            stackView.setCustomSpacing(32, after: paymentMethodForm) // custom spacing from figma
+        }
+        stackView.addArrangedSubview(updateButton)
+        stackView.addArrangedSubview(removeButton)
+        stackView.addArrangedSubview(errorLabel)
```

Comment:
I don't know how to insert footnoteLabel at an index into the stackView array— addArrangedSubview adds to the end of the array. If I add these to the array on line 50, then when I add footnoteLabel, it will show up below the remove button.

Reply:
Never mind, found out that there's insertArrangedSubview(view, at)